localstate/localstate.go

```diff
@@ -9,11 +9,25 @@ import (
 	"github.com/pkg/errors"
 )
 
-const refsDir = "refs"
+const (
+	refsDir  = "refs"
+	groupDir = "__group__"
+)
 
 type State struct {
-	LocalPath      string
+	// LocalPath is the absolute path to the context
+	LocalPath string
+	// DockerfilePath is the absolute path to the Dockerfile
 	DockerfilePath string
+	// GroupID is the ID of the state group that this ref belongs to
+	GroupID string `json:",omitempty"`
+}
+
+type StateGroup struct {
+	// Ref is the unique ID of the group ref
+	Ref string `json:"-"`
```

Comment:
We should save the configuration of the bake command. This is variables and targets (ideally connected with the build refs). Also path to the bake file(s).

I'm not sure if a full bake definition is also needed. Maybe the `--set` options?

Reply:
Yes variables, targets and overrides looks good to me. Bake definition can be useful. I'm looking at it.